File: code/backend/src/application/repositories/quarto-repository.ts
import { Quarto } from '../../domain/entities/Quarto';

export interface IQuartoRepository {
    save(quarto: Quarto): Promise<void>;
    update(quarto: Quarto): Promise<void>;
    findById(id: string): Promise<Quarto | null>;
    findByNumero(numero: string): Promise<Quarto | null>;
    findAll(): Promise<Quarto[]>;
    delete(id: string): Promise<void>;
}

File: code/stack-2/backend/src/main/java/com/hotel/repository/QuartoRepository.java
package com.hotel.repository;

import com.hotel.model.Quarto;
import org.springframework.stereotype.Repository;
import java.util.*;
import java.util.concurrent.ConcurrentHashMap;

@Repository
public class QuartoRepository implements IQuartoRepository {
    private final Map<String, Quarto> quartos = new ConcurrentHashMap<>();

    public Quarto save(Quarto quarto) {
        quartos.put(quarto.getNumero(), quarto);
        return quarto;
    }

    public Optional<Quarto> findByNumero(String numero) {
        return Optional.ofNullable(quartos.get(numero));
    }

    public List<Quarto> findAll() {
        return new ArrayList<>(quartos.values());
    }

    public void deleteByNumero(String numero) {
        quartos.remove(numero);
    }
}
